src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -311,9 +313,24 @@ public override ImmutableArray<CSharpAttributeData> GetAttributes()
 
         internal override MarshalPseudoCustomAttributeData? MarshallingInformation => _baseParameterForAttributes?.MarshallingInformation;
 
-        internal override bool IsMetadataOptional => _baseParameterForAttributes?.IsMetadataOptional == true;
+        internal override bool IsMetadataOptional => _baseParameterForAttributes is not null ? _baseParameterForAttributes.IsMetadataOptional == true : base.IsMetadataOptional;
```

Comment:
>_baseParameterForAttributes.IsMetadataOptional == true [](http://example.com/codeflow?start=95&length=54)

Is it possible that `_baseParameterForAttributes.ExplicitDefaultConstantValue == null` and `_defaultValue != null`?

Reply:
No, it's actually not, since if `_baseParameterForAttributes` is non-null, then `_baseParameterForAttributes.ExplicitDefaultConstantValue == _defaultValue`. However, we could have a situation where `baseParameterForAttributes == null`, (since it was not passed in) but `_defaultValue != null`

Reply:
Should we always attempt to pass in `baseParameterForAttributes` if possible?

Reply:
> I just checked and it shouldn't be possible, because if _baseParameterForAttributes is non-null, then _baseParameterForAttributes.ExplicitDefaultConstantValue == _defaultValue. 

Let's assert that in the constructor. That should be sufficient, thanks.